core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -160,6 +160,9 @@ class BaseUnit : INamed, IConstruction {
             && construction.cityInfo.containsBuildingUnique("All newly-trained melee, mounted, and armored units in this city receive the Drill I promotion"))
             unit.promotions.addPromotion("Drill I", isFree = true)
 
+        //movement penalty
+        if(unit.name != "Landsknecht")
+        unit.currentMovement = 0f
```

Comment:
Instead of checking by `name`, I think we should add a unique to Landsknecht and check for it here.

Reply:
ok I addded a new unique for this :)